mothernet/priors/step_function_prior.py

```diff
@@ -0,0 +1,46 @@
+import torch
+
+from mothernet.distributions import parse_distributions, sample_distributions
+from mothernet.utils import default_device
+
+
+class StepFunctionPrior:
+    def __init__(self, config=None):
+        self.config = parse_distributions(config or {})
+
+    def _get_batch(self, batch_size, n_samples, num_features, device=default_device, num_outputs=1, epoch=None,
+                  single_eval_pos=None):
+        # small wrapper to sample from the prior but also return the individual step functions.
+        with torch.no_grad():
+            hypers = sample_distributions(self.config)
+            if hypers['sampling'] == 'uniform':
+                x = torch.rand(batch_size, n_samples, num_features, device=device)
+            else:
+                x = torch.randn(batch_size, n_samples, num_features, device=device)
+
+            # Per element in batch create a random step function per feature and add them across samples.
+            start = torch.randint(0, 2, size=(batch_size, num_features), device=device).unsqueeze(dim=1)
+            step = torch.randint(0, n_samples, (batch_size, num_features), device=device)
+
+            # Expand step to have an extra dimension
+            step = step[:, :, None].expand(-1, -1, n_samples)
```

Comment:
```suggestion
            step = step.unsqueeze(-1).expand(-1, -1, n_samples)
```
I think this is the more idiomatic pytorch way? though I'm not sure. Yours is certainly the numpy way.